delfin/drivers/dell_emc/unity/unity.py

```diff
@@ -173,13 +176,378 @@ def list_alerts(self, context, query_para=None):
+                            ipv4_mask = UnityStorDriver.handle_port_ip(
+                                ipv4_mask, ip_content.get('netmask'))
+                        else:
+                            ipv6 = UnityStorDriver.handle_port_ip(
+                                ipv6, ip_content.get('ipAddress'))
+                            ipv6_mask = UnityStorDriver.handle_port_ip(
+                                ipv6_mask, ip_content.get('netmask'))
+                port_result = {
+                    'name': content.get('name'),
+                    'storage_id': self.storage_id,
+                    'native_port_id': content.get('id'),
+                    'location': content.get('name'),
+                    'connection_status': conn_status,
+                    'health_status': status,
+                    'type': constants.PortType.ETH,
+                    'logical_type': '',
+                    'max_speed': int(content.get('speed')) * units.Mi,
+                    'native_parent_id':
+                        content.get('storageProcessor').get('id'),
+                    'wwn': '',
+                    'mac_address': content.get('macAddress'),
+                    'ipv4': ipv4,
+                    'ipv4_mask': ipv4_mask,
+                    'ipv6': ipv6,
+                    'ipv6_mask': ipv6_mask
+                }
+                port_list.append(port_result)
+
+    def get_fc_port(self, port_list):
+        ports = self.rest_handler.get_all_fcports()
+        if ports is not None:
+            port_entries = ports.get('entries')
+            for port in port_entries:
+                content = port.get('content', {})
+                health_value = content.get('health', {}).get('value')
+                if health_value in UnityStorDriver.HEALTH_OK:
+                    status = constants.PortHealthStatus.NORMAL
+                else:
+                    status = constants.PortHealthStatus.ABNORMAL
+                conn_status = status
+                port_result = {
+                    'name': content.get('name'),
+                    'storage_id': self.storage_id,
+                    'native_port_id': content.get('id'),
+                    'location': content.get('slotNumber'),
+                    'connection_status': conn_status,
+                    'health_status': status,
+                    'type': constants.PortType.FC,
+                    'logical_type': '',
+                    'max_speed': int(content.get('currentSpeed')) * units.Gi,
+                    'native_parent_id':
+                        content.get('storageProcessor').get('id'),
+                    'wwn': content.get('wwn')
+                }
+                port_list.append(port_result)
 
     def list_ports(self, context):
-        pass
+        try:
+            port_list = []
+            self.get_eth_ports(port_list)
+            self.get_fc_port(port_list)
+            return port_list
```

Comment:
the code style here is not good, please follow:
```
port_list = []
port_list.extend(self.get_eth_ports())
port_list.extend(self.get_fc_ports())
return port_list
```

Reply:
done